test/parallel/test-vm-measure-memory.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const vm = require('vm');
+const {
+  SUMMARY,
+  DETAILED
+} = vm.constants.measureMemory.mode;
+
+common.expectWarning('ExperimentalWarning',
+                     'vm.measureMemory is an experimental feature. ' +
+                     'This feature could change at any time');
+
+// Test measuring memory of the current context
+{
+  vm.measureMemory(undefined)
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({})
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: SUMMARY })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: DETAILED })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  assert.throws(() => vm.measureMemory(null), {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+  assert.throws(() => vm.measureMemory('summary'), {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+  assert.throws(() => vm.measureMemory({ mode: -1 }), {
+    code: 'ERR_OUT_OF_RANGE'
+  });
+}
+
+// Test measuring memory of the sandbox
+{
+  const sandbox = vm.createContext();
```

Comment:
There's nothing that appears to cover the fail condition when `MeasureMemory` context is not fully initialized.

Reply:
I am not sure how I could create such a race?

Reply:
I'm not sure either. I'm wondering if throwing is even necessary in this case. If it's not possible to create that race from user code, would an assert be better here than a throw?

Reply:
It looks like the context could only fail due to e.g. allocation failure or maximum stack size reached, which seems difficult to test without flakes and is not what we usually test for, so I'll leave it here (but with an explanation in the doc of the error)